flow/raster_cache.cc

```diff
@@ -62,6 +62,26 @@ static bool CanRasterizePicture(SkPicture* picture) {
   return true;
 }
 
+static bool CanRasterizeDisplayList(DisplayList* display_list) {
+  if (display_list == nullptr) {
+    return false;
+  }
+
+  const SkRect cull_rect = display_list->bounds();
+
+  if (cull_rect.isEmpty()) {
+    // No point in ever rasterizing an empty display list.
+    return false;
+  }
+
+  if (!cull_rect.isFinite()) {
```

Comment:
I think we've discussed this before in another context, but it seems like it'd be a mistake to ever allow a display list to have an infinite cull rect.

In particular, this could mean that someone has introduced NaN into the cull rect, which is almost certainly going to be a programming error.

Reply:
Another piece of code stolen from the Picture cache. All of these methods in here are duplicates of the adjacent methods that take SkPicture.

If someone introduces a non-finite op then we should probably ignore the one op that did so, but not discard the picture entirely. We can at least avoid caching that picture, though.

Reply:
I think most of our Dart API surface should prevent that from happening - but if somehow it's being let through we should be letting the programmer know.

Reply:
What are you suggesting? DML_CHECK? FML_LOG?

Reply:
Log is probably good, if we can provide any details about which picture or which call did it that's ideal but I doubt we can.

Reply:
Somewhere someone was looking to add some annotation records to our "pictures". The DL could accomodate that and then it could log the last "info annotation" to provide a programmer a way to trace it back in their program.

Reply:
I added an FML_LOG(INFO) to both the Picture and DL caching methods.